src/main/java/com/carpool/car_pool/services/RideRequestService.java

```diff
@@ -126,4 +126,12 @@ public RideRequestResponse answerRideRequest(@Valid AnswerRideRequestRequest req
 
         return rideRequestConverter.entityToDTO(rideRequest);
     }
+
+    public List<RideRequestResponse> getRideRequestsForUser(UserEntity currentUser) {
+        List<RideRequestsEntity> rideRequests = rideRequestRepository.findByRequester(currentUser);
```

Comment:
Here we can get exception so you have to grab it or make the repository function Optional